src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs

```diff
@@ -100,7 +108,15 @@ private async ValueTask SelectItemAsync(CancellationToken cancellationToken)
         {
             // Switch to the UI so we can determine where the user is and determine the state the last time we updated
             // the UI.
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+#pragma warning disable VSTHRD004 // Await SwitchToMainThreadAsync
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken).NoThrowAwaitable();
```

Comment:
>SwitchToMainThreadAsync [](http://example.com/codeflow?start=56&length=23)

While this file is being modified, would it make sense to also change the SwitchToMainThreadAsync call in SelectItemWorkerAsync?

Reply:
These were limited to specific cases based on a performance trace. The other method either wasn't showing up in that trace, or didn't follow the same pattern of having an ignored return value.